src/System.Net.Sockets/src/System/Net/Sockets/UnixDomainSocketEndPoint.Unix.cs

```diff
@@ -39,7 +37,17 @@ public UnixDomainSocketEndPoint(string path)
             }
 
             _path = path;
-            _encodedPath = s_pathEncoding.GetBytes(_path);
+
+            // Pathname socket addresses should be null-terminated.
+            // Linux abstract socket addresses start with a zero byte, they must not be null-terminated.
+            bool isAbstract = path.Length > 0 && path[0] == '\0';
```

Comment:
Won't this do the wrong thing on macOS?  There it would seem the path should be interpreted as being empty, right?

Reply:
We can add some platform check here (e.g. PNSE when isAbstract && !Linux), or we can let the system calls return an error when the address is used. Currently it's the second. I can change it to the former if you want.

Reply:
> we can let the system calls return an error when the address is used

If it'll return an error that we then propagate, I'm fine with that.  We should have a test for it that runs on macOS though.